src/main/java/com/ibm/devops/connect/CloudSocketComponent.java

```diff
@@ -161,11 +162,26 @@ public void handleDelivery(String consumerTag, Envelope envelope,
                 }
             };
 
-            channel.basicConsume(queueName, true, consumer);
+            if (queueIsAvailable(channel, queueName)) {
+                log.info("Queue is available, consuming...");
+                channel.basicConsume(queueName, true, consumer);
+            } else {
+                log.info("Queue not available, waiting to attempt reconnect");
+                connectToAMQP();
+            }
         }
-        
     }
 
+    public static boolean queueIsAvailable(Channel channel, String queueName) throws IOException, InterruptedException {
+        try {
+          channel.queueDeclarePassive(queueName);
+          return true;
+        } catch (IOException e) {
+          Thread.sleep(5000);
```

Comment:
why sleep before returning?

Reply:
To give reporting consumer a few seconds to create the queue, it can take up to 30 seconds sometimes and didn't want to spam logs with messages or use more resources than is necessary